pkg/api/scylla/v1alpha1/types_monitoring.go

```diff
@@ -162,6 +162,16 @@ type ScyllaDBMonitoringSpec struct {
 
 	// components hold additional config for the monitoring components in use.
 	Components *Components `json:"components"`
+
+	// type determines what platform service model the monitoring is deployed for. Defaults to SaaS. Valid options
+	// are Platform and SaaS.
+	// "Platform" means the monitoring is deployed in the ScyllaDB cluster owner's platform. A full monitoring setup
+	// designed for cluster administrators is deployed.
+	// "SaaS" means the monitoring is deployed for a SaaS offering. A simplified monitoring setup is deployed.
+	// +kubebuilder:default:="SaaS"
+	// +kubebuilder:validation:Enum="SaaS";"Platform"
```

Comment:
```suggestion
	// type determines the deployment type of the monitoring setup. Can be "Platform" or "Saas". 
	// "Platform" sets up a detailed ScyllaDB monitoring, including the infrastructure.
	// "Saas" deploys monitoring setup focusing on the ScyllaDB service.
	// Defaults to "SaaS".
	// +kubebuilder:validation:Enum="SaaS";"Platform"
	// +kubebuilder:default:="SaaS"
```

Reply:
(i think it's better to first introduce the values and then say what the default is)

Reply:
It sort of copied it from core Service object, but I don't mind either.